keps/sig-network/4410-k8s-network-interface/README.md

```diff
@@ -0,0 +1,102 @@
+# KEP-4410: Kubernetes Networking reImagined
+
+> **NOTE**: for the initial PR we've removed a lot of the templated text and
+> aimed to keep this first iteration small and easier to consume. We are only
+> focusing on the "What" and "Why" (e.g. motivation, goals, user stories) for
+> this iteration so that we can build consensus on those first before we add
+> any of the "How".
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+<!-- /toc -->
+
+## Summary
+
+This proposal is to design and implement the KNI [Kubernetes Networking Interface] or better known as Kubernetes Networking reImagined. KNI will create a Network resource and provide an API that will provide network status, availability, how to attach a pod to a network, detach the pod from the network and update a pods network.  
+
+## Motivation
+
+Kubernetes networking has traditionally been challenging to understand for users
+interacting with the Kubernetes API, and there has been considerable flexibility
+in how Container Network Interfaces (CNIs) set up networking within clusters.
+This has resulted in a scenario where things like pod networking (including pod
+to pod networking) is opaque to users, with different implementations taking
+markedly different approaches. This fragmentation has spread networking across
+all layers of the stack which include k8s components like kube-proxy, netpol agents,
+container runtime with CNI plugins and low level runtimes like kata and issues
+with the API have negatively impacted adoption in sectors such as telecommunications.
+Our goal is to transform Kubernetes networking by making networks and their components
+actual resources within the Kubernetes API. This will allow for the development
+of shared functionalities and their integration into the API. We anticipate that
+this new approach will enhance support for areas that are currently struggling,
+facilitate the development and promotion of common features, and better define
+and accommodate advanced functionalities and potential areas for expansion.
+
+### Goals
+
+1. Design a cool looking t-shirt
+2. Provide Kubernetes APIs for the creation, configuration and management of networks (e.g. `Pod` networks)
+3. Provide documentation, examples, troubleshooting and FAQ's for KNI.
+4. Establish feature parity with current CNI [ADD, DEL]
+5. Handle support levels like Gateway API (e.g. "core" and "extended")
+6. Handle implementation-specific use cases through extension points
+7. Decouple the Pod and Node Network setup
+8. Simplify/enable triggering garbage collection to ensure no resources are left behind
```

Comment:
garbage collection of what?

Reply:
Should we include specific examples of ebpf programs, linux network artifacts such as linux bridges?

Reply:
my point was more that you are giving goals like "we will garbage collect things" without having mentioned the existence of anything that will need to be garbage-collected.